modules/consensus/synchronize.go

```diff
@@ -11,47 +11,35 @@ import (
 )
 
 const (
-	MaxCatchUpBlocks = 50
+	MaxCatchUpBlocks       = 50
+	MaxSynchronizeAttempts = 8
+	ResynchronizeTimeout   = time.Second * 20
 )
 
-// threadedResynchronize continuously calls Synchronize on a random peer every
-// few minutes. This helps prevent unintentional desychronization in the event
-// that broadcasts start failing.
+// threadedResynchronize will call synchronize on up to 8 random peers.
 func (s *State) threadedResynchronize() {
 	for {
-		go func() {
-			// The set of connected peers is small and randomly ordered, so
-			// just naively iterate through them until a Synchronize succeeds.
-			for _, peer := range s.gateway.Peers() {
-				if s.Synchronize(peer) == nil {
-					break
-				}
+		peers := s.gateway.Peers()
+		for _, peer := range peers {
+			// NOTE: error is not checked, because nothing happens whether
+			// there is an error or not, the node continuously synchronizes
+			// to all peers.
+			err := s.Synchronize(peer)
+			if err != nil {
+				continue
 			}
-		}()
-		time.Sleep(time.Minute * 2)
+			time.Sleep(ResynchronizeTimeout)
+		}
+		time.Sleep(ResynchronizeTimeout)
```

Comment:
The one thing that gives me reservations is that you want the resynchronization loop to be faster than the median block. The average block is 10m but the median is closer to 7 minutes.

I know presumably it doesn't matter but as a failsafe it's more effective that way. Is it okay if I keep it at 2 minutes?